olmo/train.py

```diff
@@ -242,7 +242,6 @@ def __post_init__(self):
 
     @property
     def dataset(self) -> IterableDataset:
-        assert isinstance(self.train_loader.dataset, IterableDataset)
```

Comment:
does this check have to be removed due to the fact that the custom dataset is now of type `IterableDataset` as well?

Reply:
Good catch, I removed these when I wasn't working with the `IterableDataset` class, I can add them back in